templates/base/packages/nextjs/tsconfig.json.template.mjs

```diff
@@ -20,9 +22,23 @@
     "plugins": [
       {
         "name": "next"
-      }
-    ]
+      },
+      ...(extraPlugins?.[0] || [])
```

Comment:
Templating in create-eth works this way: when you passing some param from args file or from template default variable, it comes to template `contents` function as an array. For example, when you passing `extraPlugins: null`, `contents` function receives `extraPlugins = [null]`.
You can read about it [here](https://github.com/scaffold-eth/create-eth/blob/main/contributors/TEMPLATING.md#template-file-contents), but that information is a bit outdated, we need to update it and add info about working with variables other than strings

So, since `extraPlugins` always has at least default value, `contents` function  always receives `extraPlugins` array. Hence, `?.` is not needed. Also, I think it's better to make default value empty array, so we can use just `...extraPlugins[0]` here

Reply:
Oh yeah I have tried doing 
   ` ...extraPlugins[0]`
and
   ` ...extraCompilerOptions[0]`
but looking at my previous changes, it looks like it didn't work because I didnot make these an empty array and an empty object, respectively.

Thank you so much for the explanation!